mesonbuild/linkers/linkers.py

```diff
@@ -1045,6 +1045,36 @@ class QualcommLLVMDynamicLinker(LLVMDynamicLinker):
     id = 'ld.qcld'
 
 
+class NAGDynamicLinker(PosixDynamicLinkerMixin, DynamicLinker):
+
+    """NAG Fortran linker, ld via gcc indirection."""
+
+    id = 'nag'
+
+    def build_rpath_args(self, env: 'Environment', build_dir: str, from_dir: str,
+                         rpath_paths: str, build_rpath: str,
+                         install_rpath: str) -> T.Tuple[T.List[str], T.Set[bytes]]:
+        if not rpath_paths and not install_rpath and not build_rpath:
+            return ([], set())
+        args = []
+        origin_placeholder = '$ORIGIN'
+        processed_rpaths = prepare_rpaths(rpath_paths, build_dir, from_dir)
+        all_paths = mesonlib.OrderedSet([os.path.join(origin_placeholder, p) for p in processed_rpaths])
+        if build_rpath != '':
+            all_paths.add(build_rpath)
+        for rp in all_paths:
+            args.extend(self._apply_prefix('-Wl,-Wl,,-rpath,,' + rp))
```

Comment:
This really needs -Wl,-Wl?

Also, isn't apply_prefix supposed to add the -Wl, for you?

Sorry I'm on my phone so reading other video is kinda hard

Reply:
Yes the nagfor compiler driver assembles its correct list of runtime objects and calls gcc as 'linker', therefore to pass options again to ld you need double indirection

"
-Wl,option
    Pass option directly to the host C compiler when linking (producing the executable). Multiple options may be specified in a single -Wl, option by separating them with commas. A comma may be included in an option by repeating it, e.g. -Wl,-filelist=file1,,file2,,file3 becomes the linker option -filelist=file1,file2,file3.
"

The compiler class doesn't have a prefix defined at the moment if I understand correctly (because of this extra indirection needed?), but if you advise that it should then I'll set it up that way.

Reply:
Ah, no that's fine. It's just an unusual setup. Rustc also uses GCC/clang as a linker, but it handles the correct prefixes itself. This probably deserves a comment as well